src/main/java/sirius/biz/jupiter/JupiterConnector.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.jupiter;
+
+import redis.clients.jedis.Client;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.exceptions.JedisConnectionException;
+import sirius.db.redis.RedisDB;
+import sirius.kernel.async.Operation;
+import sirius.kernel.commons.Watch;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.health.Microtiming;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Permits to access a <b>Jupiter</b> instance.
+ * <p>
+ * This connector is in charge of taking care about the connection management. It also permits to failover to
+ * a fallback instance in case the main instance isn't reachable.
+ * <p>
+ * If a failover is performed, we will continute to use the fallback for up to 60s before we attempt to switch
+ * back to the main instance.
+ */
+public class JupiterConnector {
+
+    private static final JupiterCommand CMD_SET_CONFIG = new JupiterCommand("SYS.SET_CONFIG");
+
+    private final String instanceName;
+    private final RedisDB redis;
+    private final RedisDB fallbackRedis;
+    private final Jupiter jupiter;
+    private long fallbackActiveUntil;
+
+    protected JupiterConnector(Jupiter jupiter, String instanceName, RedisDB redis, RedisDB fallbackRedis) {
+        this.jupiter = jupiter;
+        this.instanceName = instanceName;
+        this.redis = redis;
+        this.fallbackRedis = fallbackRedis;
+    }
+
+    /**
+     * Returns the instance (config) name of this Jupiter instance.
+     *
+     * @return the name of the config instace being used
+     */
+    public String getName() {
+        return instanceName;
+    }
+
+    /**
+     * Determines if access to Redis is configured.
+     *
+     * @return <tt>true</tt> if at least a host is available, <tt>false</tt> otherwise
+     */
+    public boolean isConfigured() {
+        return redis.isConfigured();
+    }
+
+    /**
+     * Executes one or more commands and returns a value of the given type.
+     * <p>
+     * If a HA option is present in <tt>jupiter.ha</tt>, the given fallback pool is used in case the main
+     * pool isn't available.
+     *
+     * @param description a description of the actions performed used for debugging and tracing
+     * @param task        the actual task to perform using redis
+     * @param <T>         the generic type of the result
+     * @return a result computed by <tt>task</tt>
+     */
+    public <T> T query(Supplier<String> description, Function<Client, T> task) {
+        if (fallbackRedis == null) {
+            return queryDirect(description, task);
+        }
+
+        if (fallbackActiveUntil > 0 && fallbackActiveUntil < System.currentTimeMillis()) {
+            fallbackActiveUntil = 0;
```

Comment:
Won't this trigger fallback to fallback server in else in line 95?

Reply:
The idea is to "attempt to switch back" to the main instance - if its still dead, we do and want to failover again. So during an outage of the main instance we expect 2 log messages every 60s - with the hope that the outage is short :-P

Reply:
Now I got it :-)